src/utils/IO/branchdata_checks.jl

```
@@ -59,105 +59,114 @@ function sanitize_angle_limits!(line)
-    isapprox(c, 0.0) && return Inf
-    zc = sqrt((r + im * 2 * pi * 60 * l) / (im * 2 * pi * 60 * c))
-    sil = vrated^2 / abs(zc)
-    return sil
-end
+    g = l.r / (l.r^2 + l.x^2)
+    b = -l.x / (l.r^2 + l.x^2)
+    y_mag = sqrt(g^2 + b^2)
 
-function check_sil_values(line::Union{Line, MonitoredLine}, basemva::Float64)
-    arc = get_arc(line)
-    vrated = get_base_voltage(get_to(arc))
-    SIL_levels = collect(keys(SIL_STANDARDS))
-    rating = get_rating(line)
-    closestV = findmin(abs.(SIL_levels .- vrated))
-    closestSIL = SIL_STANDARDS[SIL_levels[closestV[2]]]
-    is_valid = true
+    from_voltage_limits = get_voltage_limits(get_arc(l).from)
+    to_voltage_limits = get_voltage_limits(get_arc(l).to)
 
-    # Assuming that the rate is in pu
-    if (rating >= closestSIL.max / basemva)
-        # rate outside of expected SIL range
-        sil = calculate_sil(line, basemva)
-        @warn "Rating $(round(rating*basemva; digits=2)) MW for $(line.name) is larger than the max expected in the range of $(closestSIL)." maxlog =
-            PS_MAX_LOG
-        is_valid = false
+    fr_vmin = isnothing(from_voltage_limits) ? 0.9 : from_voltage_limits.min
+    to_vmin = isnothing(to_voltage_limits) ? 0.9 : from_voltage_limits.min
+
+    c_max = sqrt(fr_vmin^2 + to_vmin^2 - 2 * fr_vmin * to_vmin * cos(theta_max))
+    new_rate = y_mag * max(fr_vmin, to_vmin) * c_max
+
+    return new_rate
+end
+
+function correct_rate_limits!(branch::Union{Line, MonitoredLine}, basemva::Float64)
+    theoretical_line_rate_pu = line_rating_calculation(branch)
+    for field in [:rating, :rating_b, :rating_c]
+        rating_value = getfield(branch, field)
+        if isnothing(rating_value)
+            @assert field ∈ [:rating_b, :rating_c]
+            continue
+        end
+        if rating_value < 0.0
+            @error "PowerSystems does not support negative line rates. $(field): $(rating)"
+            return false
```

If we are not supporting this, shall we throw the error instead?

---

if false is returned it fails after